shared/src/actor/builtin.rs

```diff
@@ -82,5 +84,41 @@ impl TryFrom<&str> for Type {
     }
 }
 
+impl From<&Type> for String {
+    fn from(t: &Type) -> String {
+        match t {
+            Type::System => String::from("system"),
+            Type::Init => String::from("init"),
+            Type::Cron => String::from("cron"),
+            Type::Account => String::from("account"),
+            Type::Power => String::from("storagepower"),
+            Type::Miner => String::from("storageminer"),
+            Type::Market => String::from("storagemarket"),
+            Type::PaymentChannel => String::from("paymentchannel"),
+            Type::Multisig => String::from("multisig"),
+            Type::Reward => String::from("reward"),
+            Type::VerifiedRegistry => String::from("verifiedregistry"),
+        }
+    }
+}
+
 /// A mapping of builtin actor CIDs to their respective types.
 pub type Manifest = BiBTreeMap<Cid, Type>;
+
+pub fn load_manifest<B: Blockstore>(bs: &B, root_cid: &Cid) -> Result<Manifest, String> {
+    let vec: Vec<(String, Cid)> = match bs.get_cbor(root_cid) {
+        Ok(Some(vec)) => vec,
+        Ok(None) => {
+            return Err("cannot find manifest root cid".to_string());
+        }
+        Err(what) => {
+            return Err(what.to_string());
+        }
```

Comment:
Try using the `?` operator? Then we just need to deal with the Option.

Reply:
ok.

Reply:
I anyhowized it, but the second "?" needed a match. anyhow, it looks a little better now.